src/trackers/SPD.py

```
@@ -305,27 +304,33 @@ async def upload(self, meta, disctype):
-                status_message = f'data error: HTTP {e.response.status_code} - {e.response.text}'
+                meta['tracker_status'][self.tracker]['status_message'] = f'data error: HTTP {e.response.status_code} - {e.response.text}'
+                return False
             except httpx.TimeoutException:
-                status_message = f'data error: Request timed out after {self.session.timeout.write} seconds'
+                meta['tracker_status'][self.tracker]['status_message'] = f'data error: Request timed out after {self.session.timeout.write} seconds'
+                return False
             except httpx.RequestError as e:
-                status_message = f'data error: Unable to upload. Error: {e}.\nResponse: {response}'
+                meta['tracker_status'][self.tracker]['status_message'] = f'data error: Unable to upload. Error: {e}.\nResponse: {response}'
+                return False
             except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)